core/classes/IssueStatusChangeTimelineEvent.class.php

```diff
@@ -45,24 +55,67 @@ public function __construct( $p_timestamp, $p_user_id, $p_issue_id, $p_old_statu
 		$this->issue_id = $p_issue_id;
 		$this->old_status = $p_old_status;
 		$this->new_status = $p_new_status;
+		$this->type = $this->change_type();
 	}
 
 	/**
-	 * Returns html string to display
-	 * @return string
+	 * Return the type of status change
+	 * @return int One of the status change constants defined above
 	 */
-	public function html() {
+	private function change_type() {
 		$t_resolved = config_get( 'bug_resolved_status_threshold' );
 		$t_closed = config_get( 'bug_closed_status_threshold' );
 
-		if( $this->old_status < $t_closed && $this->new_status >= $t_closed ) {
-			$t_string = sprintf( lang_get( 'timeline_issue_closed' ), user_get_name( $this->user_id ), string_get_bug_view_link( $this->issue_id ) );
-		} else if( $this->old_status < $t_resolved && $this->new_status >= $t_resolved ) {
-			$t_string = sprintf( lang_get( 'timeline_issue_resolved' ), user_get_name( $this->user_id ), string_get_bug_view_link( $this->issue_id ) );
+		if( $this->old_status < $t_resolved && $this->new_status >= $t_resolved ) {
+			return IssueStatusChangeTimelineEvent::RESOLVED;
+		} else if( $this->old_status < $t_closed && $this->new_status >= $t_closed ) {
```

Comment:
In the case of an issue moving from new/assigned to closed.  Do we want to show it as "resolved" or "closed"?  In my implementation I was going for "closed", with your updates, it will show up as "resolved".